app/models/evaluator_submission_assignment.rb

```diff
@@ -22,25 +22,47 @@ class EvaluatorSubmissionAssignment < ApplicationRecord
     assigned: 0,
     unassigned: 1,
     recused: 2,
+    recused_unassigned: 3
+  }
+
+  def evaluation_status
+    if assigned?
+      if evaluation.nil?
+        :not_started
+      elsif evaluation.completed_at.nil?
+        :in_progress
+      else
+        :completed
+      end
+    else
+      status.to_sym
+    end
+  end
+
+  ORDER_VALUES = {
+    recused: 0,
+    unassigned: 1,
+    recused_unassigned: 2,
     not_started: 3,
     in_progress: 4,
-    completed: 5,
-    recused_unassigned: 6
-  }
+    completed: 5
+  }.freeze
+
+  def ordering_priority
+    case evaluation_status
+    when :recused then ORDER_VALUES[:recused]
+    when :unassigned then ORDER_VALUES[:unassigned]
+    when :recused_unassigned then ORDER_VALUES[:recused_unassigned]
+    when :not_started then ORDER_VALUES[:not_started]
+    when :in_progress then ORDER_VALUES[:in_progress]
+    when :completed then ORDER_VALUES[:completed]
+    end
+  end
 
   scope :ordered_by_status, lambda {
-    order(
-      Arel.sql(
-        "CASE evaluator_submission_assignments.status
-          WHEN #{ActiveRecord::Base.connection.quote(statuses[:recused])} THEN 0
-          WHEN #{ActiveRecord::Base.connection.quote(statuses[:unassigned])} THEN 1
-          WHEN #{ActiveRecord::Base.connection.quote(statuses[:recused_unassigned])} THEN 2
-          WHEN #{ActiveRecord::Base.connection.quote(statuses[:not_started])} THEN 3
-          WHEN #{ActiveRecord::Base.connection.quote(statuses[:in_progress])} THEN 4
-          WHEN #{ActiveRecord::Base.connection.quote(statuses[:completed])} THEN 5
-          ELSE 6
-        END"
-      )
-    )
+    select('evaluator_submission_assignments.*, evaluations.id AS evaluation_id, evaluations.completed_at').
+      left_joins(:evaluation).
+      to_a.
+      sort_by(&:ordering_priority)
```

Comment:
since this is now sorting in memory instead of in the query, it isn't a scope any more. it should be a normal method on the class. [scopes should be chainable with other scopes](https://andycroll.com/ruby/a-scope-should-return-a-scope/), but this lambda returns an array which means it can't be used with other scopes.